### PR TITLE
Introduce new property "X5t" in X509SecurityKey

### PR DESCRIPTION
**Pull request by ChenYan98**

### CODE REVIEW

**Comment by ChenYan98** [comment]
#454

**Comment by PinpointTownes** [comment]
Hum, this change breaks the existing layering.

`x5t` is a JWK-specific property and I'm not sure it's a good idea to add it to `X509SecurityKey`, which is part of `Microsoft.IdentityModel.Tokens`. Maybe it would be better to simply expose a `byte[]` `Thumbprint` property and make the base64url-encoding in the JWT handler.

**Comment by brentschmaltz** [comment]
I understand the layering, but following that will result in a base64-url encoding each time we want to match a key to an x5t. I don't think there is anything wrong with offering this property.

File: src/Microsoft.IdentityModel.Tokens/X509SecurityKey.cs [hunk 1]
```diff
@@ -54,6 +54,7 @@ public X509SecurityKey(X509Certificate2 certificate)
 
             _certificate = certificate;
             KeyId = certificate.Thumbprint;
+            X5t = Base64UrlEncoder.Encode(certificate.Thumbprint);
```
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Tokens/X509SecurityKey.cs]: I think this should be Base64UrlEnoder.Encode(certificate.GetCertHash()) according to https://tools.ietf.org/html/rfc7517#section-4.8

**Pull request merged**